editor/tests/org.fusesource.ide.projecttemplates.tests.integration/src/main/java/org/fusesource/ide/projecttemplates/tests/integration/wizards/FuseIntegrationProjectCreatorRunnableIT.java

```diff
@@ -273,27 +266,28 @@ public void executeOnFailure() {
 			}
 		});
 		executePomAction.launch(getSelectionForLaunch(project), ILaunchManager.DEBUG_MODE);
+		launch = executePomAction.getLaunch();
+		
+		
 		int currentAwaitedTime = 0;
 		while (currentAwaitedTime < ThreadGarbageCollector.THREAD_LIFE_DURATION && !deploymentFinished) {
 			readAndDispatch(0);
-			try (JMXConnector jmxc = JMXConnectorFactory.connect(new JMXServiceURL(ICamelDebugConstants.DEFAULT_JMX_URI))) {
-				MBeanServerConnection mbsc = jmxc.getMBeanServerConnection();
-				deploymentFinished = !mbsc.queryMBeans(new ObjectName(CamelDebugFacade.CAMEL_DEBUGGER_MBEAN_DEFAULT), null).isEmpty();
-				isDeploymentOk = deploymentFinished;
-				System.out.println("JMX connection succeeded");
-				System.out.println("isDeployment Finished? " + isDeploymentOk);
-			} catch(IOException ioe){
-				System.out.println("JMX connection attempt failed");
+			System.out.println("isDeployment Finished? " + deploymentFinished);
+			Optional<IDebugTarget> camelDebugTarget = Stream.of(launch.getDebugTargets())
+				.filter(debugTarget -> debugTarget instanceof CamelDebugTarget)
+				.findAny();
+			if(camelDebugTarget.isPresent()) {
+				MBeanServerConnection mBeanConnection = ((CamelDebugTarget) camelDebugTarget.get()).getMBeanConnection();
+				isDeploymentOk = mBeanConnection != null
+						&& !mBeanConnection.queryMBeans(new ObjectName(CamelDebugFacade.CAMEL_DEBUGGER_MBEAN_DEFAULT), null).isEmpty();
+				if(isDeploymentOk) {
+					deploymentFinished = true;
+				}
 			}
 			Thread.sleep(500);
 			currentAwaitedTime += 500;
 		}
 		assertThat(isDeploymentOk).as("build/deployment failed, you can have a look to the file "+ mavenOutputFilePath + " for more information.").isTrue();
-		launch = executePomAction.getLaunch();
-		assertThat(Stream.of(launch.getDebugTargets())
-				.filter(debugTarget -> debugTarget instanceof CamelDebugTarget)
-				.collect(Collectors.toList()))
-		.isNotEmpty();
```

Comment:
removed as it is now part of the requirements to detect finished deployments few lines above